llpc/tool/llpcGraphicsPipelineBuilder.h

```diff
@@ -0,0 +1,51 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2022 Google LLC. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcGraphicsPipelineBuilder.h
+ * @brief LLPC header file: graphics pipeline compilation logic for standalone LLPC compilers.
+ ***********************************************************************************************************************
+ */
+#pragma once
+
+#include "llpcPipelineBuilder.h"
+
+namespace Llpc {
+namespace StandaloneCompiler {
+
+// Pipeline builder implementation for graphics pipelines.
+class GraphicsPipelineBuilder : public PipelineBuilder {
+public:
+  using PipelineBuilder::PipelineBuilder;
```

Comment:
nit: It seems like `GraphicsPipelineBuilder` and `ComputePipelineBuilder` are declared similarly. Are the declarations expected to diverge in the future or would it make sense to move the shared parts of the declaration and the implementation into a common parent class?

Reply:
`Graphics` and `ComputePipelineBuilder` have a similar interface but different compilation logic. In the future, the interfaces may diverge. I don't think having a common base class will make the code easier to understand. The `build` function will have to be overridden anyway to do something different.

Reply:
Thank you for clarification.